_posts/2019-03-23-WassersteinGAN.markdown

```diff
@@ -0,0 +1,522 @@
+      </p>
+      </details>
+
+  2. Describe in your own words the difference between a generative and a discriminative model.
+      <details><summary>Solution</summary>
+      <p>
+      This is an open ended question but here are some of the differences:
+      <ul>
+        <li>In the generative setting we usually model \(p(x)\), our models are usually non-deterministic, and we can sample from them.</li>
+        <li>In the discriminative setting we usually model \(p(y|x)\), our models are often deterministic, and we can't necessarily sample from them.</li>
+      </ul>
+      </p>
+      </details>
+ 
+  _These last two questions are a good barometer for determining your understanding of the challenges involved in training generative models._ 
+
+  3. Theis _et al._’s paper claims “a model with zero KL divergence will produce perfect samples” &#8212; why is this the case?
+      <details><summary>Solution</summary>
+      <p>
+      As we showed last week, \(D_{KL}(p||q) = 0\) if and only if \(p(x)\), the true data distribution, and \(q(x)\) the model distribution, are the same. 
+      </p>
+      <p>
+      Therefore, if \(D_{KL}(p||q) = 0\) samples from our model will be indistinguishable from the real data.
+      </p>
+      </details>
+
+  4. Explain why the high log-likelihood of a generative model might not correspond to realistic samples?
+     <details><summary>Solution</summary>
+      <p>
+      Theis <i>et al.</i> outlined two scenarios where this is the case:
+      <ul>      
+      <li><b>Low likelihood & good samples</b>: our model can overfit to the training data and produce good samples, however, because the model has overfit it will have a low likelihood for unseen test data.</li>
+      <li><b>High likelihood & poor samples</b>: here the issue is that high dimensional data will tend to have higher log-likelihoods than low dimensional data. </li>
+      </ul>
+      </p>
+      </details>
+
+  **Notes**: Here is a [link](todo:add_link) to our notes for the lesson. We were fortunate enough to have Tim Salimans sit in on the session!
+
+  <br />
+
+# 3 Generative Adversarial Networks
+
+  **Motivation**: Let’s read the original GAN paper. Our main goal this week is to understand how GANs solve some of the problems with training generative models, as well as, some of the new issues that come with training GANs.
+
+  _This week has a fairly heavy workload and could plausibly be split over two weeks. This is due to having two papers for required reading. However, it should be noted that the second paper is really somewhere between required and optional &#8212; we think that it contains some  interesting material and sets the state well for looking at WGAN in week 4, but the most important points are covered again in week 4. Depending on your study group you might want to focus on this paper more or less based on your interest (we recommend that most groups don't spend too much time on this paper)._
+
```

Comment:
Again, this comment was a bit tricky. Don't want to discourage people from reading the second paper, it is very interesting and everyone from the group who read it got pretty engrossed by it. However, as a result of people getting a little sidetracked the week became a bit overloaded so I wanted to give some kind of warning.

Reply:
As opposed to the first comment, which is just highlighting that the first week has fairly basic (yet important fundamental) stuff, I think this comment is pretty fair. You might be able to make this comment less verbose and just highlight the second paper is optional, but highly recommended? Curious to hear your thoughts.

Reply:
I agree that it was a bit too verbose. Let me know what you think of the changes I made.